### PR TITLE
Box support for Indexer endpoints

### PR DESCRIPTION
**Pull request by algochoi**
This PR uses the existing generator to generate paths for the indexer box/boxes endpoint and adds cucumber unit tests to test correct paths.

### CODE REVIEW

File: test/indexer_unit_test.go [hunk 1]
```diff
@@ -55,6 +55,24 @@ func IndexerUnitTestContext(s *godog.Suite) {
 	s.Step(`^we make a Search Accounts call with exclude "([^"]*)"$`, weMakeASearchAccountsCallWithExclude)
 	s.Step(`^we make a Lookup Account by ID call against account "([^"]*)" with exclude "([^"]*)"$`, weMakeALookupAccountByIDCallAgainstAccountWithExclude)
 	s.Step(`^we make a SearchForApplications call with creator "([^"]*)"$`, weMakeASearchForApplicationsCallWithCreator)
+	s.Step(`^we make a LookupApplicationBoxByIDandName call with applicationID (\d+) with encoded box name "([^"]*)"$`,
```
Comment (algochoi) [inline test/indexer_unit_test.go:58]: This is the only non-generated file

File: test/docker/run_docker.sh [hunk 2]
```diff
@@ -5,7 +5,7 @@ set -e
 # reset test harness
 rm -rf test-harness
 rm -rf test/features
-git clone --single-branch --branch feature/box-storage https://github.com/algorand/algorand-sdk-testing.git test-harness
+git clone --single-branch --branch indexer-boxes https://github.com/algorand/algorand-sdk-testing.git test-harness
```
Comment (algochoi) [inline test/docker/run_docker.sh:8]: TODO: Revert back before merging

**Review by ahangsu — APPROVED** [review]

File: test/indexer_unit_test.go [hunk 3]
```diff
@@ -55,6 +55,24 @@ func IndexerUnitTestContext(s *godog.Suite) {
...
+		func(appId int, encodedBoxName string) error {
+			indexerClient, err := indexer.MakeClient(mockServer.URL, "")
+			if err != nil {
+				return err
+			}
+			indexerClient.LookupApplicationBoxByIDandName(uint64(appId)).Name(encodedBoxName).Do(context.Background())
+			return nil
+		})
+	s.Step(`^we make a SearchForApplicationBoxes call with applicationID (\d+) with max (\d+)$`,
```
Comment (tzaffi) [inline test/indexer_unit_test.go:67]: I think this should also include a `next "([^"]*)"$`. Also, I'm not familiar enough with this repo to understand if the same `next` encoding problem I [hilighted in the python SDK](https://github.com/algorand/py-algorand-sdk/pull/366/files#r948447135), applies here.

Comment (ahangsu) [reply test/indexer_unit_test.go]: good point, just add the segment for `next`.

File: test/docker/run_docker.sh (hunk 2 above)
Comment (tzaffi) [reply test/docker/run_docker.sh:8]: This file should no longer exist

**Review by tzaffi — CHANGES_REQUESTED** [review]
seems like the most recent changes haven't been merged into the `indexer-boxes` branch.

File: .test-env [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Configs for testing repo download:
 SDK_TESTING_URL="https://github.com/algorand/algorand-sdk-testing"
-SDK_TESTING_BRANCH="feature/box-storage"
+SDK_TESTING_BRANCH="indexer-boxes"
```
Comment (ahangsu) [inline .test-env:3]: TODO: revert back once algorand-sdk-testing PR merge
```suggestion
SDK_TESTING_BRANCH="feature/box-storage"
```

**Review by tzaffi — APPROVED** [review]

**Pull request merged**